src/qililab/instrument_controllers/instrument_controller.py

```diff
@@ -254,9 +254,12 @@ def __str__(self):
 
     def to_dict(self):
         """Return a dict representation of the InstrumentReference class."""
-        return {
+        controller_dict = {
             RUNCARD.NAME: self.name.value,
             RUNCARD.ALIAS: self.alias,
             INSTRUMENTCONTROLLER.CONNECTION: self.connection.to_dict(),
             INSTRUMENTCONTROLLER.MODULES: [module.to_dict() for module in self.settings.modules],
         }
+        if not self.settings.reset:
```

Comment:
You are only saving the reset, when its `False`, is that because the default is `True` and it's not necessary in such case?

Reply:
That is the reason yes. First to not overload the runcard with default parameters, as we are already not doing as we are not saving all the parameters but only those that might change, and also not to have to change many unittests to fix this bug.